### ISSUE TITLE
Problem with solution in CVXPY in v1.0.31 returns unbounded in v1.1.0

### ISSUE THREAD

**Issue by kmonson** [cvxgrp/cvxpy#1263]
1.0.31 produces a solution and 1.1.0 reports the problem as unbounded. (It produces the same result with any of the 1.1.* versions.)

It looks like it's generating very different problems for the solver when I change versions.

A and G are different dimensions, dims is different, h is different, and the inverse data is very different (only start shown). c is identical.

Version 1.0.31:
```
[{'A': <8761x35040 sparse matrix of type '<class 'numpy.float64'>'
	with 35040 stored elements in Compressed Sparse Column format>,
  'G': <123005x35040 sparse matrix of type '<class 'numpy.float64'>'
	with 175198 stored elements in Compressed Sparse Column format>,
  'b': array([-0., -0., -0., ..., -0., -0., -0.]),
  'c': array([-36.73, -36.47, -36.57, ...,   0.  ,   0.  ,   0.  ]),
  'dims': (zero: 8761, nonpos: 123005, exp: 0, soc: [], psd: []),
  'h': array([ -0.,  -0.,  -0., ..., 800., 800., 800.]),
  'offset': array([0.])},
 Chain(reductions=[<cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x12c3953d0>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x12c395580>, <cvxpy.reductions.eval_params.EvalParams object at 0x12c395c70>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x12c3917f0>, <cvxpy.reductions.solvers.conic_solvers.ecos_conif.ECOS object at 0x12b0c8910>]),
 [<cvxpy.reductions.inverse_data.InverseData object at 0x12c391190>,
  [{0: ('var7186', Variable((8760,)), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None)},
   {0: ('pv_out', Variable((8760,), nonneg=True), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None)},
   {30: 30,
    34: 34,
    38: 38,
```

Version 1.1.0:
```
[{'A': <8761x43800 sparse matrix of type '<class 'numpy.float64'>'
	with 35040 stored elements in Compressed Sparse Column format>,
  'G': <26280x43800 sparse matrix of type '<class 'numpy.float64'>'
	with 26280 stored elements in Compressed Sparse Column format>,
  'b': array([0., 0., 0., ..., 0., 0., 0.]),
  'c': array([-36.73, -36.47, -36.57, ...,   0.  ,   0.  ,   0.  ]),
  'dims': (zero: 8761, nonneg: 26280, exp: 0, soc: [], psd: []),
  'h': array([0., 0., 0., ..., 0., 0., 0.]),
  'param_prob': <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ParamConeProg object at 0x122d3a880>},
 Chain(reductions=[<cvxpy.reductions.dcp2cone.dcp2cone.Dcp2Cone object at 0x121e152b0>, <cvxpy.reductions.cvx_attr2constr.CvxAttr2Constr object at 0x121e154f0>, <cvxpy.reductions.eval_params.EvalParams object at 0x121e0f160>, <cvxpy.reductions.dcp2cone.cone_matrix_stuffing.ConeMatrixStuffing object at 0x121e0f850>, <cvxpy.reductions.solvers.conic_solvers.ecos_conif.ECOS object at 0x120e07ca0>]),
 [<cvxpy.reductions.inverse_data.InverseData object at 0x121e0b340>,
  [{0: ('var0', Variable((8760,)), None),
    1: ('ene', Variable((8760,)), None),
    2: ('dis', Variable((8760,)), None),
    3: ('ch', Variable((8760,)), None),
    4299: ('var4299', Variable((24,)), None),
    4319: ('var4319', Variable((24,)), None),
    4339: ('var4339', Variable((24,)), None),
    4359: ('var4359', Variable((24,)), None),
    4379: ('var4379', Variable((24,)), None),
    4399: ('var4399', Variable((24,)), None),
```

**To Reproduce**
Data files and code to reproduce this can be found here https://github.com/kmonson/cvxpy-1.1-bug.
```
git clone https://github.com/kmonson/cvxpy-1.1-bug
cd cvxpy-1.1-bug
pip install -r requirements.txt
./test.sh
```

`test.sh` will dump info about the created problems to `1.0.31.txt` and `1.1.0.txt` respectively.

**Expected behavior**
1.1.0 should produce the same output (except for floating point error) as 1.0.31

**Version**
 - OS: macOS Catalina
 - CVXPY Version: 1.1.0

**Comment by SteveDiamond** [comment]
This is a real mystery! I looked into it and am not really sure what the cause is. @rileyjmurray did we make a switch from NonPos to NonNeg at some point in the 1.1 upgrade for the conic standard form? There are a lot more (slack?) variables in the 1.1 output. Of course that shouldn't affect the correctness.

**Comment by SteveDiamond** [comment]
Probably for debugging purposes we should have some way to record the output of every stage of the reduction chain.

**Comment by SteveDiamond** [comment]
Also @kmonson really nice job setting up this debugging example! We'll figure this out eventually.

**Comment by kmonson** [comment]
@SteveDiamond Thanks! 

I messed around with it earlier this week and, unfortunately, I couldn't find any parts of the problem I could strip out and maintain the same behavior for both versions of cvxpy. We might be able to reduce the size of the problem down from 8760 to 24-48 just to make it less unwieldily, but I ran into problems last time I tried that. I'll give it another shot and let you know.

**Comment by kmonson** [comment]
@SteveDiamond I was able to truncate the data to one day's worth and it still manifested the problem. I just pushed it out to the bug demonstration repro. It reduces the problem size dramatically.

**Comment by SteveDiamond** [comment]
Thanks!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SteveDiamond** [comment]
Definitely the switch from nonpositive to nonnegative as the default constraint has something to do with it. The strangest thing to me is that in 1.1 the vector `h` is all zeros. That seems wrong.

**Comment by SteveDiamond** [comment]
Ok the reason you're seeing this is because you use ``NonNeg`` and ``NonPos`` directly instead of the normal <= constraints. Something is wrong with the canonicalization of ``NonNeg`` and ``NonPos``.

**Comment by kmonson** [comment]
Ah, that gives me a path forward to try to fix it on my end. I'll mess around with it and see if it can be made to work.

**Comment by kmonson** [comment]
I'll leave the code as is in the bug demo repo, naturally.

**Comment by kmonson** [comment]
Removing all calls to Zero and NonPos and replacing them with equivalent operators fixes the issue. The results are identical (ignoring floating point error). I'm going to make some updates to the bug demo repo to facilitate validating a fix for the original problem.

**Comment by kmonson** [comment]
The updated version of the bug demo repo is up. 

It now captures the result in a csv file if it was optimal for each run.

It now runs 4 cases: 
1.0.31 with and without using Zero and NonPos (mostly so I can ensure that my switch to using operators was done correctly)
1.1.0 with and without using Zero and NonPos.

Replace `pip install cvxpy==1.1.0` in `test`.sh` to install a fixed version. If it's working correctly then 1.1.0.csv should be within floating point error the other 3 csv files.

**Comment by SteveDiamond** [comment]
Fixed by #1277
